lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1423,6 +1405,36 @@ def write_store(
         )
         return rval
 
+    @router.delete(
+        "/api/workflows/{workflow_id}",
+        summary="Add the deleted flag to a workflow.",
+    )
+    def delete_workflow(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        workflow_id: DecodedDatabaseIdField = StoredWorkflowIDPathParam,
+    ):
+        workflow_to_delete = self.workflows_manager.get_stored_workflow(trans, workflow_id)
+        if workflow_to_delete.user != trans.user and not trans.user_is_admin:
+            raise exceptions.InsufficientPermissionsException()
```

Comment:
```suggestion
        workflow_to_delete = self.workflows_manager.get_stored_workflow(trans, workflow_id, check_ownership=True)
```

Reply:
Am I missing something or is the `check_ownership` param not supported here?